src/java/main/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -61,17 +64,17 @@ public static void initialize() {
         }
     }
 
-    public static AuthenticationProvider getProvider(String scheme) {
+    public static ServerAuthenticationProvider getProvider(ZooKeeperServer zks, String scheme) {
```

Comment:
Yeah, I agree that loose typing can lead to errors. My concern is that we haven't needed to pass a `ZooKeeperServer` object for the previous `AuthenticationProvider` implementations, and with the current changes all calls to `getProvider` and `initialize` get such an instance. A metadata map is possibly a better way to go.